Feature/post-note

dain7672 opened the pull request:
merge 한 걸 이대로 올려도 되려나요? 확인부탁드려요~

dain7672 committed:
application-local.properties update

dain7672 committed:
Merge branch 'develop' of https://github.com/wwh-techcamp-2018/team1-WannaOne into develop

# Conflicts:
#	src/main/resources/application-local.properties

dain7672 committed:
Merge remote-tracking branch 'origin/develop' into develop

dain7672 committed:
Implements Note posting feature backend
최소의 기능만 먼저구현
아무나 post 가능.

dain7672 committed:
Implement note posting feature FE
글 저장 하기 기능 : 항상 새롭게 추가됨.
가장 기본적인 posting 기능만 추가.

kyucumber commented on src/main/java/com/wannaone/woowanote/domain/Note.java:
```diff
@@ -14,6 +15,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Getter
+@Setter
```
setter를 사용하는 부분이 없는데 달아준 이유가 있나요??

dain7672 replied:
지금은 필요한데, NoteDTO로 바꾸면 지워야 겠네요.

kyucumber reviewed (approved):
👍

doooyeon commented on src/main/java/com/wannaone/woowanote/web/ApiNoteController.java:
```diff
@@ -15,8 +15,17 @@
+    @PostMapping
+    public ResponseEntity post(@RequestBody Note note) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(noteRepository.save(note).getId());
+    }
```
노트를 save 할 때 service에서 예외 처리 할 건 없으려나..?
보통 controller에서는 service를 가지고 service에서 repository를 가지는게 일반적인데..

dooho-h replied:
그러게. 

그리고 새로운 노트 작성 뿐만 아니라 기존 노트 수정도 post 요청으로 들어올텐데 메소드 이름이 post인건 좋지 않은 것 같아요.

doooyeon reviewed (approved):
👍

kyucumber reviewed (approved):
👍

dain7672 committed:
Reset application-local.properties

dooho-h commented on src/main/resources/static/js/note.js:
```diff
@@ -42,6 +46,34 @@ class Note {
+                    headers: {'content-type': 'application/json'},
+                    body: JSON.stringify({title: title,
+                            text: text,
+                            registerDatetime: 1534219200000, //TODO: change to NoteDTO. to delete registerDate here.
```
registerDateTime을 현재시간으로 하고, updateDateTime을 null로 보내야 하는거 아닌가요?

dain7672 replied:
update 와 post를 구분시키게 되면 고려해보도록 할게요.!

dain7672 committed:
Refactor: add service and test for note posting feature #27
- Note.java : add constructor
- NoteSerivce, NoteServiceTest: add postNewNote function
- ApiNoteController: change to use service

dain7672 committed:
Refactor
현재시간이 registerDateTime에 들어가도록.
이후 update함수가 추가되면 다른 함수를 만들어주는 것으로 할 수도 있을 듯.

dooho-h reviewed (approved):
너무 수고 많았어요 💪
